flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/sharedbuffer/SharedBuffer.java

```diff
@@ -161,40 +100,30 @@ public void init(
 		eventsCount.putAll(maxIds);
 	}
 
-	/**
-	 * Stores given value (value + timestamp) under the given state. It assigns a preceding element
-	 * relation to the previous entry.
-	 *
-	 * @param stateName      name of the state that the event should be assigned to
-	 * @param eventId        unique id of event assigned by this SharedBuffer
-	 * @param previousNodeId id of previous entry (might be null if start of new run)
-	 * @param version        Version of the previous relation
-	 * @return assigned id of this element
-	 * @throws Exception Thrown if the system cannot access the state.
-	 */
-	public NodeId put(
-			final String stateName,
-			final EventId eventId,
-			@Nullable final NodeId previousNodeId,
-			final DeweyNumber version) throws Exception {
+	public SharedBufferAccessor<V> getAccessor() {
+		return new SharedBufferAccessor<>(this);
+	}
 
-		if (previousNodeId != null) {
-			lockNode(previousNodeId);
+	public void advanceTime(long timestamp) throws Exception {
```

Comment:
This could be package protected.

Reply:
This need to be called in nfa package, I think it should be public.